posts/tests.py

```diff
@@ -226,3 +236,52 @@ def test_rating_invalid_args(
     # Check if the Rating average function works properly
     def test_rating_avg_function(self, persist_rating):
         assert persist_rating.get_rating_average() == 5
+
+
+# -------------------------------------------- Poll tests --------------------------------------------
+
+
+@pytest.mark.django_db()
+class TestPoll:
+
+    # Check that the Poll values are the same as the Poll inputs.
+    def test_new_poll_input_same_as_output(self, new_poll):
+        assert new_poll.author.username == USERNAME
+        assert new_poll.description == DESCRIPTION
+
+    # Check if the Poll is saved in the database.
+    def test_persist_poll(self, persist_poll):
+        assert persist_poll in Poll.objects.all()
+
+    # Check if the Poll can be referred as post.
+    def test_inheritance_of_poll_from_post(self, persist_poll):
+        assert Post.objects.filter(poll=persist_poll).exists()
+        post = Post.objects.all().first()
+        assert hasattr(post, 'poll')
+        assert not hasattr(post, 'resume')
+
+    # Check if Poll deletion delete only Poll from database.
+    def test_delete_poll(self, persist_poll):
+        assert persist_poll in Poll.objects.all()
+        assert persist_poll.author in User.objects.all()
+        persist_poll.delete()
+        assert persist_poll not in Poll.objects.all()
+        assert persist_poll.author in User.objects.all()
+
+    # Check if Poll's author deletion delete both the author and the poll.
+    def test_delete_polls_author(self, persist_poll):
+        assert persist_poll in Poll.objects.all()
+        assert persist_poll.author in User.objects.all()
+        persist_poll.author.delete()
+        assert persist_poll not in Poll.objects.all()
+        assert persist_poll.author not in User.objects.all()
+
+    # verify that it is impossible to create a Poll with invalid parameters.
+    @pytest.mark.parametrize("user, description, expected_error", [(
+        USERNAME,
+        DESCRIPTION,
+        TypeError),
+        (new_user, 6, TypeError), (EMAIL, DESCRIPTION, TypeError)])
+    def test_poll_invalid_args(self, user, description, expected_error):
```

Comment:
You don't need to test all these variations of values \
The test is for not been able to create a `Poll` object with `user` and `description` \
the values are not relevant here, one test should cover it.

Reply:
Yes, as you said I meant to test it with `Poll(author=user, description=description)`, my bad.
I'll fix it 👍

Reply:
fixed